SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+
+    NSMutableArray<SDLTransportSelectionBox *> *secondaryTransports = nil;
+    if (payload.secondaryTransports != nil) {
+        secondaryTransports = [NSMutableArray array];
+        for (NSString *transportString in payload.secondaryTransports) {
+            SDLTransportSelection transport = [self sdl_convertTransportSelection:transportString];
+            [secondaryTransports addObject:@(transport)];
+        }
+    }
+    NSArray<SDLTransportClassBox *> *transportsForAudio = [self sdl_convertServiceTransports:payload.audioServiceTransports];
+    NSArray<SDLTransportClassBox *> *transportsForVideo = [self sdl_convertServiceTransports:payload.videoServiceTransports];
+
+    SDLLogV(@"Secondary transports: %@, transports for audio: %@, transports for video: %@", secondaryTransports, transportsForAudio, transportsForVideo);
+
+    dispatch_async(_stateMachineQueue, ^{
+        [self sdl_configureManager:secondaryTransports transportsForAudio:transportsForAudio transportsForVideo:transportsForVideo];
+    });
+}
+
+- (SDLTransportSelection)sdl_convertTransportSelection:(NSString *)transportString {
+    if ([transportString isEqualToString:@"TCP_WIFI"]) {
+        return SDLTransportSelectionTCP;
+    } else if ([transportString isEqualToString:@"IAP_BLUETOOTH"] ||
+               [transportString isEqualToString:@"IAP_USB"] ||
+               [transportString isEqualToString:@"IAP_USB_HOST_MODE"] ||
+               [transportString isEqualToString:@"IAP_USB_DEVICE_MODE"] ||
+               [transportString isEqualToString:@"IAP_CARPLAY"]) {
+        return SDLTransportSelectionIAP;
+    } else {
+        // "SPP_BLUETOOTH" and "AOA_USB" are not used
+        return SDLTransportSelectionDisabled;
+    }
+}
+
+- (nullable NSArray<SDLTransportClassBox *> *)sdl_convertServiceTransports:(nullable NSArray<NSNumber *> *)transports {
+    if (transports == nil) {
+        return nil;
+    }
+
+    // Actually there is nothing to "convert" here. We just check the range and recreate an array.
+    NSMutableArray<SDLTransportClassBox *> *array = [NSMutableArray new];
+    for (NSNumber *num in transports) {
+        int transport = [num intValue];
+        if (transport == 1) {
+            [array addObject:@(SDLTransportClassPrimary)];
+        } else if (transport == 2) {
+            [array addObject:@(SDLTransportClassSecondary)];
+        } else {
+            SDLLogV(@"Unknown transport class received: %d", transport);
+        }
+    }
+
+    return [array copy];
+}
+
+- (SDLTransportSelection)sdl_getTransportSelectionFromProtocol:(SDLProtocol *)protocol {
+    if ([protocol.transport isMemberOfClass:[SDLIAPTransport class]]) {
+        return SDLTransportSelectionIAP;
+    } else if ([protocol.transport isMemberOfClass:[SDLTCPTransport class]]) {
+        return SDLTransportSelectionTCP;
+    } else {
+        SDLLogD(@"Unknown type of primary transport");
```

Comment:
Should be an error log

Reply:
Fixed in commit 5009bc7f0137147a9211bbf3725e2dee7b5cba98.